### PR TITLE
React-react2-week1

### PR DESCRIPTION
**Pull request by alina-kho**
Homework is done :) Comments and suggestions are highly appreaciated

### CODE REVIEW

File: react/react2/week1/github-search/src/components/InputBar.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useContext } from "react";
+import { SearchContext } from "./SearchContext";
+
+export const InputBar = () => {
+  const contextValue = useContext(SearchContext);
+
+  const inputHandler = (e) => {
+    // console.log("Input handler works");
+    // console.log(contextValue.inputValue);
+    contextValue.setInputValue(e.target.value);
```
Comment (Divyajg) [inline react/react2/week1/github-search/src/components/InputBar.js:10]: `setInputValue `is neither defined anywhere in this file nor passed as input to this function!

File: react/react2/week1/github-search/src/components/InputBar.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+  return (
+    <div className="inputBar">
+      <input
+        type="text"
+        onChange={inputHandler}
+        placeholder="Search for user"
+      ></input>
+    </div>
```
Comment (Divyajg) [inline react/react2/week1/github-search/src/components/InputBar.js:20]: This file name coulde be a better one, for Ex: searchBar, UserSearch, etc.

File: react/react2/week1/github-search/src/components/SearchContext.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { createContext } from "react";
+
+export const SearchContext = createContext();
+export const SearchContextProvider = SearchContext.Provider;
```
Comment (Divyajg) [inline react/react2/week1/github-search/src/components/SearchContext.js:4]: You have just created the context, neither children are passed to context nor values are passed to children!

**Review by Divyajg — CHANGES_REQUESTED** [review]
You have very good coding talent, But need to understand a bit more about the [Context ](https://www.reactjstutorials.com/react-context/4/react-create-context) concept,
For this homework using Context is the main goal I think,
1. Define the Context component
2. Pass the children components through context Provider in the App.js file 
3. Data from the Context component is passed as values to the children
4.  That data is used in the Children components as `useContext`.
5.  Children components can be search and render components for this application.

I can understand React is difficult concept. Take your time and learn the concepts clearly and ask for help if you need anybody and anytime.

File: react/react2/week1/github-search/src/components/InputBar.js (hunk 1 above)
Comment (alina-kho) [reply react/react2/week1/github-search/src/components/InputBar.js:10]: it is defined in the App.js file and passed in contextValues (they are defined - same file, line 47)

File: react/react2/week1/github-search/src/components/SearchContext.js (hunk 3 above)
Comment (alina-kho) [reply react/react2/week1/github-search/src/components/SearchContext.js:4]: the context values object is defined in the App.js and passed as a value using contextProvider

File: react/react2/week1/github-search/src/components/InputBar.js (hunk 2 above)
Comment (alina-kho) [reply react/react2/week1/github-search/src/components/InputBar.js:20]: Good point - gonna fix it :)

File: react/react2/week1/github-search/src/components/SearchContext.js (hunk 3 above)
Comment (Divyajg) [reply react/react2/week1/github-search/src/components/SearchContext.js:4]: When you have created a component for context, You can define the context and pass values in the same component.

Comment (Divyajg) [reply react/react2/week1/github-search/src/components/SearchContext.js:4]: Your code flow might work for now as this is a very simple assignment. The advantage of using React is its components, where we can build the small components separately and connect them through the App.js file.
Going ahead, no fetching code or any logical code is included in the app.js file, it is used to connect all the components in the order we want.

**Comment by Divyajg** [comment]
This [link](https://www.reactjstutorials.com/react-context/4/react-create-context) should give you clear understanding about the concept.

**Pull request merged**